sycl/include/sycl/ext/intel/experimental/esimd.hpp

```diff
@@ -0,0 +1,26 @@
+//==------------ esimd.hpp - DPC++ Explicit SIMD API -----------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// The main header of the Explicit SIMD API.
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#ifdef __SYCL_DEVICE_ONLY__
+#define SYCL_ESIMD_KERNEL __attribute__((sycl_explicit_simd))
+#define SYCL_ESIMD_FUNCTION __attribute__((sycl_explicit_simd))
+#else
+#define SYCL_ESIMD_KERNEL
+#define SYCL_ESIMD_FUNCTION
```

Comment:
Is it possible to undef these macro at the end of the file?

Reply:
The intent is actually make this available to user code - to mark ESIMD kernels and functions.

Reply:
Could you please tell if this is documented then?

Reply:
Not, it is not. What is documented is the `sycl_explicit_simd` attribute, to which SYCL_ESIMD* macros are basically defined.

Reply:
I acknowledge the problem wit documentation, but this will be solved in a separate PR. This one does not introduce these macros - it just moves from the deprecated header to the new one.